### PR TITLE
api: return error when create grant/evict leader scheduler

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/8759

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
Fix an issue that pd can't report error normally when creating grant/evict leader scheduler.
```

### CODE REVIEW

File: server/api/scheduler.go [hunk 1]
```diff
@@ -142,10 +142,23 @@ func (h *schedulerHandler) CreateScheduler(w http.ResponseWriter, r *http.Reques
...
+		}
+		exist, err := h.AddEvictOrGrant(storeID, name)
+		if err != nil {
+			h.r.JSON(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		// we should ensure whether it is the first time to create evict-leader-scheduler
+		// or just update the evict-leader.
+		if exist {
+			h.r.JSON(w, http.StatusOK, "The scheduler has been applied to the store.")
```
Comment (lhy1024) [inline server/api/scheduler.go:159]: Do we need add a test for it?

Comment (okJiang) [reply server/api/scheduler.go:159]: I can add a test to master, and cp it to this pr

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**